// ng2-recaptcha/recaptcha/recaptcha.module.js
"use strict";
var core_1 = require('@angular/core');
var forms_1 = require('@angular/forms');
var recaptcha_common_module_1 = require('./recaptcha-common.module');
var recaptcha_loader_service_1 = require('./recaptcha-loader.service');
var recaptcha_value_accessor_directive_1 = require('./recaptcha-value-accessor.directive');
var recaptcha_component_1 = require('./recaptcha.component');
var RecaptchaModule = (function () {
    function RecaptchaModule() {
    }
    RecaptchaModule.forRoot = function () {
        return {
            ngModule: RecaptchaModule,
            providers: [recaptcha_loader_service_1.RecaptchaLoaderService],
        };
    };
    RecaptchaModule.decorators = [
        { type: core_1.NgModule, args: [{
                    declarations: [
                        recaptcha_value_accessor_directive_1.RecaptchaValueAccessorDirective,
                    ],
                    exports: [recaptcha_component_1.RecaptchaComponent, recaptcha_value_accessor_directive_1.RecaptchaValueAccessorDirective],
                    imports: [forms_1.FormsModule, recaptcha_common_module_1.RecaptchaCommonModule],
                },] },
    ];
    /** @nocollapse */
    RecaptchaModule.ctorParameters = [];
    return RecaptchaModule;
}());
exports.RecaptchaModule = RecaptchaModule;
